.tekton/scripts/check-partner-tasks.sh

```diff
@@ -0,0 +1,159 @@
+                    if [ ! -s "$readme_file" ]; then
+                        echo "warning: it is recommended to provide an informative document for task $task_name in $readme_file" >>"$check_result"
+                    fi
+                fi
+            fi
+        done
+    done
+
+    if [ -s "$check_result" ]; then
+        cat "$check_result"
+        echo
+    fi
+
+    [ -s "$resultf" ] && return 1
+    return 0
+}
+
+check_privilege_use() {
+    resultf=$(mktemp)
+    check_result=$(mktemp)
+
+    check_cases="\
+allowPrivilegeEscalation:true:allowPrivilegeEscalation is not allowed to be true to request more privileges.
+runAsUser:0:root(0) is not allowed to be set to securityContext.runAsUser.
+privileged:true:securityContext.privileged is not allowed to be true."
+
+    find partners/*/*/*.yaml | awk -F '/' '{ print $0, $2, $4 }' | \
+    while read -r task_file task_name yaml_file; do
+        if [ "${task_name}.yaml" != "$yaml_file" ]; then
+            continue
+        fi
+
+        while IFS=: read -r sc_config check_value err_msg; do
+            task_idx=0
+            # When no securityContext config is set, yq outputs message "Error: no matches found" to stderr.
+            # Redirect this message into stdout in order to suppress these potential Errors, and such error
+            # messages does not match any config value obviously.
+            for value in $(yq -e ".spec.steps[].securityContext.$sc_config" "$task_file" 2>&1); do
+                if [ "$value" == "$check_value" ]; then
+                    echo "found" >>"$resultf"
+                    step_name=$(yq -e ".spec.steps[$task_idx].name" "$task_file")
+                    echo "error: in step $step_name, $err_msg" >>"$check_result"
+                fi
+                task_idx=$((task_idx+1))
+            done
+        done <<<"$check_cases"
+
+        if [ -s "$check_result" ]; then
+            echo "Task $task_file has unexpected privileges:"
+            cat "$check_result"
+            echo
+            echo >"$check_result"
+        fi
+    done
+
+    [ -s "$resultf" ] && return 1
+    return 0
+}
+
+check_task_schema() {
+    resultf=$(mktemp)
+    check_result=$(mktemp)
+    find partners/*/*/*.yaml | awk -F '/' '{ print $0, $2, $4 }' | \
+    while read -r task_file task_name yaml_file; do
+        if [ "${task_name}.yaml" != "$yaml_file" ]; then
+            continue
+        fi
+
+        if ! oc apply -f "$task_file" --dry-run=server >"$check_result" 2>&1; then
+            echo "issue found" >>"$resultf"
+            echo "Task schema validation failed: $task_file"
+            cat "$check_result"
+            echo
+        fi
+
+        echo >"$check_result"
+    done
+    [ -s "$resultf" ] && return 1
+    return 0
+}
+
+exitcode=0
+check_dir_structure_status=Fail
+check_task_schema_status="n/a "
+check_privilege_use_status="n/a "
+
+if check_dir_structure; then
+    check_dir_structure_status=Pass
+    if check_task_schema; then
```

Comment:
for manual execution it could run `oc whoami` -> if non-zero return code then it's not logged and check_task_schema should be skipped.

Reply:
Done